pkg/controller/datasource-controller.go

```diff
@@ -102,8 +103,22 @@ func (r *DataSourceReconciler) update(ctx context.Context, dataSource *cdiv1.Dat
 		if isReady {
 			updateDataSourceCondition(dataSource, cdiv1.DataSourceReady, corev1.ConditionTrue, "DataSource is ready to be consumed", ready)
 		}
+	} else if sourceSnapshot := dataSource.Spec.Source.Snapshot; sourceSnapshot != nil {
```

Comment:
consider refactoring the PVC/Snapshot source functionality by extracting it to 2 funcs.

Reply:
Would it be possible to do in a follow-up?

Reply:
done